pandas/tslib.pyx

```diff
@@ -1908,6 +1909,17 @@ cpdef array_to_datetime(ndarray[object] values, raise_=False,
                     iresult[i] = value
                     _check_dts_bounds(&dts)
                 except ValueError:
+                    # if requiring iso8601 strings, skip trying other formats
+                    if require_iso8601:
+                        if coerce:
```

Comment:
I think you can just use ``format is not None``. its passed (but not used) anyhow, rather than add another parameter?